### PR TITLE
📊 Under-fifteen mortality rate - two versions of UN IGME

### PR DESCRIPTION
**Pull request by spoonerf**
Calculating under-fifteen mortality rates using a combination of two vintages of the UN IGME dataset. The older vintage (2018) has data that is no longer available in the live UN IGME dataset.

We also calculate under-fifteen mortality from under-five mortality and 5-14 years mortality.

This indicator is available [here](http://staging-site-fiona/admin/variables/812042) and will be used to update [this chart](https://ourworldindata.org/grapher/youth-mortality-rate).

### CODE REVIEW

File: etl/steps/data/garden/un/2023-08-16/igme.py [hunk 1]
```diff
@@ -58,6 +88,125 @@ def run(dest_dir: str) -> None:
...
+    )
+    tb = tb.drop(columns="adjusted_5_14_mortality_rate")
+    return tb
+
+
+def combine_datasets(tb_a: Table, tb_b: Table, table_name: str, preferred_source: str) -> Table:
+    """
+    Combine two tables with a preference for one source of data.
+    """
+    tb_combined = pr.concat([tb_a, tb_b]).sort_values(["country", "year", "source"])
```
Comment (lucasrodes) [inline etl/steps/data/garden/un/2023-08-16/igme.py:123]: you can set the `short_name` directly from the `pr.concat` call

File: etl/steps/data/garden/un/2023-08-16/igme.py [hunk 2]
```diff
@@ -58,6 +88,125 @@ def run(dest_dir: str) -> None:
...
+
+    return tb_combined
+
+
+def remove_duplicates(tb: Table, preferred_source: str, dimensions: List[str]) -> Table:
+    """
+    Removing rows where there are overlapping years with a preference for IGME data.
+
+    """
+    assert any(tb["source"] == preferred_source)
```
Comment (lucasrodes) [inline etl/steps/data/garden/un/2023-08-16/igme.py:140]: could add an error message

File: etl/steps/data/garden/un/2023-08-16/igme.py [hunk 3]
```diff
@@ -58,6 +88,125 @@ def run(dest_dir: str) -> None:
...
+
+    tb_no_duplicates = tb[~duplicate_rows]
+
+    tb_duplicates = tb[duplicate_rows]
+
+    tb_duplicates_removed = tb_duplicates[tb_duplicates["source"] == preferred_source]
+
+    tb = pr.concat([tb_no_duplicates, tb_duplicates_removed])
+
+    assert len(tb[tb.duplicated(subset=dimensions, keep=False)]) == 0
```
Comment (lucasrodes) [inline etl/steps/data/garden/un/2023-08-16/igme.py:152]: could add an error message

File: etl/steps/data/garden/un/2023-08-16/igme.py [hunk 4]
```diff
@@ -58,6 +88,125 @@ def run(dest_dir: str) -> None:
...
+                "Observation value": "five_to_fourteen_mortality",
+                "Lower bound": "five_to_fourteen_mortality_lb",
+                "Upper bound": "five_to_fourteen_mortality_ub",
+            }
+        )
+    )
+    tb_u15 = pr.merge(tb_u5, tb_5_14)
+    tb_u15["under_fifteen_mortality"] = tb_u15["under_five_mortality"] + tb_u15["five_to_fourteen_mortality"]
+    tb_u15["under_fifteen_mortality_lb"] = tb_u15["under_five_mortality_lb"] + tb_u15["five_to_fourteen_mortality_lb"]
+    tb_u15["under_fifteen_mortality_ub"] = tb_u15["under_five_mortality_ub"] + tb_u15["five_to_fourteen_mortality_ub"]
```
Comment (lucasrodes) [inline etl/steps/data/garden/un/2023-08-16/igme.py:186]: can rewrite

```python
for suffix in ["", "_lb", "_ub"]:
    tb_u15[f"under_fifteen_mortality{suffix}"] = tb_u15[f"under_five_mortality{suffix}"] + tb_u15[f"five_to_fourteen_mortality{suffix}"]
```

File: dag/main.yml [hunk 5]
```diff
@@ -559,6 +549,7 @@ steps:
     - snapshot://un/2023-08-16/igme.zip
   data://garden/un/2023-08-16/igme:
     - data://meadow/un/2023-08-16/igme
+    - data://garden/un/2023-09-26/igme
```
[the remark this answers is not in the document]
Comment (spoonerf) [reply dag/main.yml:552]: That's a good point - the version dates are currently just the dates I added them to etl, which is quite unhelpful. I'll update this to more reflect the dataset version e.g. year of release would be better for the older dataset.

File: etl/steps/data/garden/un/2023-08-16/igme.meta.yml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+tables:
+  igme_combined:
+    variables:
+      ? observation_value_deaths_per_1_000_live_births_under_fifteen_mortality_rate_both_sexes_all_wealth_quintiles
+      : title: Under fifteen mortality rate, both sexes, all wealth quintiles
```
[the remark this answers is not in the document]
Comment (spoonerf) [reply etl/steps/data/garden/un/2023-08-16/igme.meta.yml:5]: I'm not sure! VScode automatically added it for me for indicators where they length is >70 characters

File: dag/main.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (lucasrodes) [reply dag/main.yml:552]: I see ー That makes sense, thanks!

File: etl/steps/data/garden/un/2023-09-26/igme.countries.json [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+  "United States of America": "United States",
+  "Uruguay": "Uruguay",
+  "Uzbekistan": "Uzbekistan",
+  "Vanuatu": "Vanuatu",
+  "Venezuela (Bolivarian Republic of)": "Venezuela",
+  "Viet Nam": "Vietnam",
+  "Yemen": "Yemen",
+  "Zambia": "Zambia",
+  "Zimbabwe": "Zimbabwe",
+  "The former Yugoslav Republic of Macedonia": "North Macedonia"
```
[the remark this answers is not in the document]
Comment (spoonerf) [reply etl/steps/data/garden/un/2023-09-26/igme.countries.json]: Good point - Gapminder maintains the values for just one entity name over the entire time-period - they say a bit about it here: https://www.gapminder.org/data/geo/changes/. I think we're okay to do the above harmonization.

File: etl/steps/data/garden/un/2023-08-16/igme.py (hunk 4 above)
Comment (spoonerf) [reply etl/steps/data/garden/un/2023-08-16/igme.py:186]: Oh nice one, thanks! ✨

**Pull request merged**